src/rest/middlewares/errorHandler.ts

```diff
@@ -0,0 +1,30 @@
+import { FastifyReply, FastifyRequest } from "fastify";
+import { inject, singleton } from "tsyringe";
+import { ApplicationError } from "../../application/errors/ApplicationError";
+import { Logger } from "../../tools/logger";
+import { HttpError } from "../errors/HttpError";
+
+@singleton()
+export class ErrorHandlerMiddleware {
+  constructor(@inject(Logger) private logger: Logger) {
+    this.handleError = this.handleError.bind(this);
+  }
+
+  async handleError(
```

Comment:
The async is not useful here :)
```suggestion
  handleError(
```